elcid/assets/js/elcid/controllers/add_antifungal_patients.js

```diff
@@ -0,0 +1,144 @@
+angular.module('opal.controllers').controller("AddAntifungalPatients", function($scope, DemographicsSearch){
+  "use strict";
+
+  $scope.patientForms = [];
+
+  $scope.states = {
+    EDITING: "editing",
+    SEARCHING: "searching",
+    FOUND: "found"
+  }
+
+  $scope.stages = {
+    LOOKUP_PATIENTS: "lookup_patients",
+    ADD_PATIENTS: "add_patients"
+  }
+
+  var patient = {
+    state: $scope.states.EDITING,
+    error: false,
+    demographics: {
+      hospital_number: undefined,
+      // if populated from the server this has the rest of demographcis.to_dict()
+    },
+  }
+
+  $scope.flattenAndClean = function(hospitalNumbers){
+    /*
+    * takes in a string of hospital numbers seperated by
+    * new line, spaces and or commas.
+    *
+    * returns an an array of hospital numbers
+    */
+    var splitted = _.map(hospitalNumbers.split(" "), function(x){
+      return _.map(x.split(","), function(y){
+        return _.map(y.split("\n"))
+      })
+    })
+    return _.uniq(_.compact(_.flatten(splitted)));
+  }
+
+
+  $scope.lookupAll = function(){
+    $scope.stage = $scope.stages.ADD_PATIENTS;
+    var hns = $scope.flattenAndClean($scope.editing.hospitalNumbers);
+    $scope.patientForms = _.map(hns, newPatientFromHospitalNumer);
+  }
+
+  var newPatientFromHospitalNumer = function(hospitalNumber){
+    /*
+    * Return a new patientForm and start the search
+    */
+   var newPatient = _.clone(patient);
+   newPatient.demographics = {hospital_number: hospitalNumber};
+   $scope.lookup(newPatient);
+   return newPatient;
+  }
+
+  $scope.getSearchArgs = function(patientForm){
+    var searchArgs = {}
+    searchArgs[DemographicsSearch.PATIENT_FOUND_IN_ELCID] = function(result){
+      patientForm.demographics = result.demographics[0];
+      patientForm.state = $scope.states.FOUND;
+      patientForm.error = false;
+    }
+    searchArgs[DemographicsSearch.PATIENT_FOUND_UPSTREAM] = function(result){
+      patientForm.demographics = result.demographics[0];
+      patientForm.state = $scope.states.FOUND;
+      patientForm.error = false;
+    }
+    searchArgs[DemographicsSearch.PATIENT_NOT_FOUND] = function(){
+      patientForm.state = $scope.states.EDITING;
+      patientForm.error = true;
+    }
+
+    return searchArgs;
+  }
+
+
+  $scope.getDemographicsJson = function(){
+    return JSON.stringify(_.pluck($scope.patientForms, "demographics"));
+  }
+
+  $scope.lookup = function(patientForm){
+    /*
+    * Query and handle the response of looking up
+    * the hospital number for a patient Form.
+    */
+    if(!patientForm.demographics.hospital_number || !patientForm.demographics.hospital_number.length){
+      return
+    }
+    patientForm.error = undefined;
+    patientForm.state = $scope.states.SEARCHING;
+    var args = $scope.getSearchArgs(patientForm);
+    DemographicsSearch.find(patientForm.demographics.hospital_number, args);
+  }
+
+  $scope.research = function(){
```

Comment:
Can we rename to e.g. `re_search` to avoid it being read as "research() and development()`